File: frontend/src/types/case.ts
export type CaseStatus = 'open' | 'investigating' | 'closed';
export type CaseSeverity = 'low' | 'medium' | 'high' | 'critical';

export interface Case {
  id: string;
  title: string;
  description?: string;
  status: CaseStatus;
  severity: CaseSeverity;
  created_at: string;
  updated_at: string;
  assigned_to?: string;
  email_ids?: string[];
  notes?: CaseNote[];
}

export interface CaseNote {
  id: string;
  case_id?: string;
  author: string;
  content: string;
  created_at: string;
}

export interface CaseTimelineItem {
  type: 'case_created' | 'email_linked' | 'note_added' | 'audit_event' | string;
  timestamp: string;
  actor?: string;
  title: string;
  details?: any;
  email_id?: string;
  note_id?: string;
  action?: string;
  severity?: string;
}

export interface CaseFilterParams {
  status?: CaseStatus;
  severity?: CaseSeverity;
  limit?: number;
  offset?: number;
}

export interface CreateCaseInput {
  title: string;
  description: string;
  severity?: CaseSeverity;
  assigned_to?: string;
}

export interface UpdateCaseInput {
  title?: string;
  description?: string;
  status?: CaseStatus;
  severity?: CaseSeverity;
  assigned_to?: string;
}

export interface CreateCaseNoteInput {
  content: string;
  author?: string;
}

File: backend/app/schemas/case.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime
from uuid import UUID
from app.models.email_case import CaseStatus, CaseSeverity

class CaseCreate(BaseModel):
    title: str
    description: str
    severity: CaseSeverity

class CaseUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[CaseStatus] = None
    severity: Optional[CaseSeverity] = None
    assigned_to: Optional[str] = None

class CaseNoteCreate(BaseModel):
    content: str
    author: str

class CaseNoteResponse(BaseModel):
    id: UUID
    author: str
    content: str
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class CaseResponse(BaseModel):
    id: UUID
    title: str
    description: str
    status: CaseStatus
    severity: CaseSeverity
    created_at: datetime
    updated_at: datetime
    assigned_to: Optional[str]
    email_ids: List[UUID]
    notes: List[CaseNoteResponse]
    model_config = ConfigDict(from_attributes=True)
